src/lambda.rs

```diff
@@ -0,0 +1,94 @@
+mod commp;
+
+use std::error::Error;
+use std::str::FromStr;
+
+#[macro_use]
+extern crate lambda_runtime as lambda;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate log;
+extern crate simple_logger;
+
+use lambda::error::HandlerError;
+
+use rusoto_core::credential::{AwsCredentials, StaticProvider};
+use rusoto_core::region::Region;
+use rusoto_s3::{GetObjectRequest, S3Client, S3};
+
+use hex;
+
+#[derive(Deserialize, Clone)]
+struct CommPRequest {
+    region: String,
+    bucket: String,
+    key: String,
+}
+
+#[derive(Serialize, Clone)]
+struct CommPResponse {
+    region: String,
+    bucket: String,
+    key: String,
+    commp: String,
+    size: u64,
+    #[serde(rename = "paddedSize")]
+    padded_size: u64,
+    #[serde(rename = "pieceSize")]
+    piece_size: u64,
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    simple_logger::init_with_level(log::Level::Info)?;
+    lambda!(commp_handler);
+
+    Ok(())
+}
+
+fn commp_handler(
+    request: CommPRequest,
+    _c: lambda::Context,
+) -> Result<CommPResponse, HandlerError> {
+    info!(
+        "Received request: {}/{}/{}",
+        request.region, request.bucket, request.key
+    );
+
+    let region = Region::from_str(request.region.as_str()).unwrap();
+
+    let client = S3Client::new_with(
+        rusoto_core::request::HttpClient::new().expect("Failed to creat HTTP client"),
+        StaticProvider::from(AwsCredentials::default()),
+        region,
+    );
+
+    let get_req = GetObjectRequest {
+        bucket: request.bucket.to_owned(),
+        key: request.key.to_owned(),
+        ..Default::default()
+    };
+
+    let result = client
+        .get_object(get_req)
+        .sync()
+        .expect("Couldn't GET object");
```

Comment:
I think here it's OK, but generally spoken it's better to propagate errors up rather than panicking.

Reply:
easier said than done! error handling and propagation is one thing I can't get a grip of yet, this use of `anyhow` seems to complicate things (I'm only pulling it in because some of rust-fil-proofs dabbles with it). `expect`, `unwrap`, `?`, are still fairly opaque to me in where they should, can and can't be used.